Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -46,13 +47,13 @@ struct ClangModuleBuildMetadata {
     let module: ClangModule
 
     /// Path to working directiory.
-    let prefix: String
+    let prefix: AbsolutePath
 
     /// Extra arguments to append to basic compile and link args.
     let otherArgs: [String]
 
     /// Path to build directory for this module.
-    var buildDirectory: String { return Path.join(prefix, "\(module.c99name).build") }
+    var buildDirectory: AbsolutePath { return prefix.appending(module.c99name + ".build") }
```

Comment:
Makes sense.  In that case I think we also want to add API to check for validity, returning a standardized and actionable set of errors.  In other words, a client that takes a name and wants to construct an AbsolutePath should not need to get involved with decisions about what makes a name valid.